src/api/app/models/bs_request.rb

```diff
@@ -714,6 +714,21 @@ def calculate_state_from_reviews
     end
   end
 
+  def find_review_for_opts(opts)
+    reviews.reverse_each do |review|
+      return review if review.reviewable_by?(opts)
+    end
+    nil
+  end
+
+  def supersede_request(history_arguments, superseded_opt)
+    self.state = :superseded
+    self.superseded_by = superseded_opt
+    history_arguments[:description_extension] = superseded_by.to_s
+    save!
```

Comment:
I am nitpicking, but you could just use [update!](https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update-21) here.